evalml/pipelines/utils.py

```diff
@@ -234,20 +234,18 @@ def _get_decomposer(X, y, problem_type, estimator_class, sampler_name=None):
     components = []
     if is_time_series(problem_type) and is_regression(problem_type):
         time_index = get_time_index(X, y, None)
-        # If the time index frequency is uninferrable, STL will fail
-        if time_index.freq is None:
-            return components
-        freq = time_index.freq.name
-        if STLDecomposer.is_freq_valid(freq):
-            # Make sure there's a seasonal period
-            order = 3 if "Q" in freq else 5
-            seasonal_period = STLDecomposer.determine_periodicity(
-                X,
-                y,
-                rel_max_order=order,
-            )
-            if seasonal_period is not None and seasonal_period <= DECOMPOSER_PERIOD_CAP:
-                components.append(STLDecomposer)
+        if time_index.freq is not None:
+            order = 3 if "Q" in time_index.freq.name else 5
+        else:
+            order = 5
+        # Make sure there's a seasonal period
+        seasonal_period = STLDecomposer.determine_periodicity(
+            X,
+            y,
+            rel_max_order=order,
+        )
+        if seasonal_period is not None and seasonal_period <= DECOMPOSER_PERIOD_CAP:
```

Comment:
The way `determine_periodicity` is set up, we're currently detecting a "period" on the single stacked target data column. I'm worried that that's too brittle, it could cause weird issues in the future. Could you put this in a conditional branch to ensure we only run it in the single series case, and for now just always add the decomposer for multiseries? We'll have to come back and revisit, but that should be ok for the MVP.